packages/terra-table/src/TableRow.jsx

```diff
@@ -19,17 +20,38 @@ const propTypes = {
    * Whether or not row is selectable
    */
   isSelectable: PropTypes.bool,
+  /**
+   * The associated metaData to be provided in the onSelect callback.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  metaData: PropTypes.object,
+  /**
+   * Function callback for when the appropriate click or key action is performed.
+   * Callback contains the javascript evnt and prop metadata, e.g. onSelect(event, metaData)
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Function callback for the ref of the tr.
+   */
+  refCallback: PropTypes.func,
 };
 
 const defaultProps = {
   isSelected: false,
-  isSelectable: undefined,
+  isSelectable: false,
 };
 
 const TableRow = ({
   children,
   isSelected,
   isSelectable,
+  metaData,
+  onBlur,
+  onClick,
+  onKeyDown,
+  onMouseDown,
```

Comment:
do we need to add onBlur, onClick, onKeyDown, and onMouseDown as first class props?